workers/udp.go

```diff
@@ -0,0 +1,130 @@
+package workers
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/frightenedmonkey/statsproxy/common"
+	"github.com/frightenedmonkey/statsproxy/config"
+	"net"
+	"strconv"
+	"strings"
+	"sync/atomic"
+)
+
+type Packet struct {
+	Bucket string
+	Raw    string
+}
+
+func StartUDPListeners(l *net.UDPConn) {
+	for i := 0; i < config.Service.Workers; i++ {
+		go udpListener(l)
+	}
+}
+
+func StartUDPWriters() {
+	for i := 0; i < config.Service.WriterMultiplier; i++ {
+		for _, host := range config.Service.Statsd.Hosts {
+			go hostWriter(host)
+		}
+	}
+}
+
+func hostWriter(host config.Host) {
+	coalescedPacket := make([]byte, 0)
+	for {
+		newMessage := <-common.HostChans[host]
+		if len(coalescedPacket)+len([]byte(newMessage)) > 1400 {
```

Comment:
1400 would be a good candidate for making it a constant; don't really know the importance of this value here